packages/integration-tests/index.js

```diff
@@ -677,8 +677,11 @@ async function waitForAllTestSf(
   while (timeWaitedSecs < maxWaitTimeSecs && workflowExecutions.length < numExecutions) {
     await delay(waitPeriodMs);
     timeWaitedSecs = (moment.duration(moment().diff(startTime)).asSeconds());
-    const executions = await getExecutions(workflowArn, 100);
-    // Search all recent executions for target payload
+    const sfExecutions = await getExecutions(workflowArn, 100);
+    const executions = sfExecutions.filter(
+      (sfExecution) => sfExecution.startDate.getTime() > Date.now() - 12 * 3600 * 1000
+    );
```

Comment:
Interesting. Has that been happening where we fetch executions for a workflow and it throws because there are a bunch of old executions associated? Maybe if you're re-running tests on a branch that hasn't been re-deployed in a while? I guess even then the workflow ARN wouldn't change so maybe any time you fetch executions that workflow could have old ones 🤔 

Great catch!

Reply:
StepFunctions.describeExecution throws error if the execution is expired, but the execution is still listed.